localstack/services/cloudformation/models/ec2.py

```diff
@@ -315,6 +319,25 @@ def fetch_state(self, stack_name, resources):
 
     @staticmethod
     def get_deploy_templates():
+        def _pre_delete(resource_id, resources, resource_type, func, stack_name):
+            res = resources[resource_id]
+            state = res[
+                KEY_RESOURCE_STATE
+            ]  # TODO: there is probably a better way to get the state here
```

Comment:
@whummer maybe you have some input here, not sure if this is very idiomatic. Might not matter too much in light of the upcoming state/lifecycle refactoring, but would love to stay consistent. :+1:

Reply:
One thing we could do is get an object representation of the resource, then extract the `.state` property:

```
    @classmethod
    def get_deploy_templates(cls):
        def _pre_delete(resource_id, resources, resource_type, func, stack_name):
            resource = cls(resources[resource_id])
            state = resource.state
            ...
```

Would be probably be a bit nicer for now. As you mentioned, in a future iteration I'd expect that the `cls(resources[resource_id])` conversion will go away, once we start passing proper objects around instead of dicts.